src/notification_batcher.py

```
@@ -512,5 +535,209 @@ def _populate_list_embed(
+    if not guilds_to_broadcast:
+        return
+
+    if len(guilds_to_broadcast) == len(list(bot.guilds)):
+        await broadcast_embed_to_guilds(bot, embed, context)
+        return
+
+    for idx, guild in enumerate(guilds_to_broadcast):
+        try:
+            await send_announcement(guild, embed)
+            logger.info("Sent %s to guild %s.", context, guild.id)
+        except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)